packages/discount-code-exporter/test/main.spec.js

```diff
@@ -1,11 +1,220 @@
+        expect(result).toBeUndefined()
+        spy.mockRestore()
+        done()
+      })
+      codeExport.run(outputStream)
+    })
+
+    it('should emit error if it occurs when streaming to json', (done) => {
+      const spy = jest
+        .spyOn(codeExport, '_fetchCodes')
+        .mockImplementation(() => Promise.reject(new Error('error occured')))
+      const outputStream = streamtest['v2'].toText((error, result) => {
+        expect(error.message).toBe('error occured')
+        expect(result).toBeUndefined()
+        spy.mockRestore()
+        done()
+      })
+      codeExport.run(outputStream)
+    })
+  })
+
+  describe('::_fetchCodes', () => {
+    let processMock
+    const sampleResult = {
+      body: {
+        results: [],
+      },
+    }
+    beforeEach(() => {
+      processMock = jest.fn((request, processFn) => (
+        processFn(sampleResult).then(() => Promise.resolve())
+      ))
+    })
+
+    it('should fail if status code is not 200', async () => {
+      codeExport.client.process = processMock
+      sampleResult.message = 'Error occured'
+      try {
+        await codeExport._fetchCodes()
+      } catch (error) {
+        expect(error.message).toBe('Error occured')
+      }
+    })
+
+    it('should fetch discount codes using `process` method', async () => {
+      codeExport.client.process = processMock
+      sampleResult.statusCode = 200
+      await codeExport._fetchCodes()
+      expect(processMock).toHaveBeenCalledTimes(1)
+      expect(processMock.mock.calls[0][0])
+      .toEqual({
+        uri: '/asafaelhn/discount-codes?limit=500',
+        method: 'GET',
+      })
+    })
+
+    it('should loop over discount codes and write to stream', async () => {
+      codeExport.client.process = processMock
+      sampleResult.statusCode = 200
+      sampleResult.body.results = [ 'code1', 'code2', 'code3' ]
+      const fakeStream = { write: jest.fn() }
+      await codeExport._fetchCodes(fakeStream)
+      expect(fakeStream.write).toHaveBeenCalledTimes(3)
+    })
+  })
+
+  describe('::_buildRequest', () => {
+    it('should build request according to query', () => {
+      codeExport.predicate = 'code-predicate'
+      codeExport.accessToken = 'myAccessToken'
+      const expected = {
+        uri: '/asafaelhn/discount-codes?where=code-predicate&limit=500',
+        method: 'GET',
+        headers: {
+          Authorization: 'Bearer myAccessToken',
+        },
+      }
+      const actual = codeExport._buildRequest()
+      expect(actual).toEqual(expected)
+    })
   })
 
-  it('should return the sum of two numbers', () => {
-    expect(findSum(3, 5)).toBe(8)
+  describe('::_processCode', () => {
+    const sampleCodeObj = {
+      name: { en: 'English', de: 'German' },
+      cartDiscounts: [{
+        typeId: 'cart-discount',
+        id: 'discount-id-1',
+      }],
+    }
+
+    it('should flatten object and return `cartDiscounts` as id string', () => {
```

Comment:
No. It should return only the id of the `cartDiscount` as a string, I don't know how to write it more succinct without `max-len` error